### PR TITLE
add support for zircote/swagger-php version ^3

### LINKED ISSUE #4: Add support for OpenAPI v3
Add compatiblity with v3 of https://github.com/zircote/swagger-php. In bast case it should be compatible with v2 and v3 at the same time without any outside change needed.

### PR DESCRIPTION
**Pull request by kswzr**
Relates #4 
Closes #4

### CODE REVIEW

**Comment by Tobion** [comment]
Thanks for working on this @kswzr. I would really prefer to have a single version that is compatible with both swagger 2 and 3. The reason is that there shouldn't be a bc break that warrants a new major version. The change from swagger 2 to 3 should be seamless.
Can you implement it like this @kswzr ?

**Comment by kswzr** [comment]
Would implement it, but also preffering to bump to a new major version.

**Comment by Tobion** [comment]
I would accept it in one version as it makes maintainance easier. The tests need to be run against both open api versions.

File: README.md [hunk 1]
```diff
@@ -16,19 +16,19 @@ This way you do not have to duplicate any routing information in Symfony. Consid
 [zircote/swagger-php](https://github.com/zircote/swagger-php) like the following example:
 
 ```php
-use Swagger\Annotations as SWG;
+use Openapi\Annotations as OA;
```
Comment (Tobion) [inline README.md:19]: the casing is wrong. the namespace is called `OpenApi`. it seems not correct in several places.

File: src/OpenApiRouteLoader.php [hunk 2]
```diff
@@ -94,18 +97,18 @@ private function createRoute(Operation $operation, string $controller, FormatSuf
                 $route->setRequirement('_format', $formatSuffixConfig->pattern);
             }
         }
-        if (null !== $operation->parameters) {
+        if (\Openapi\UNDEFINED !== $operation->parameters) {
             foreach ($operation->parameters as $parameter) {
-                if ('path' === $parameter->in && null !== $parameter->pattern) {
-                    $route->setRequirement($parameter->name, $parameter->pattern);
+                if ('path' === $parameter->in && \Openapi\UNDEFINED !== $parameter->schema->pattern) {
```
Comment (Tobion) [inline src/OpenApiRouteLoader.php:102]: it seems `$parameter->schema` must also be checked for `\Openapi\UNDEFINED`

File: composer.json [hunk 3]
```diff
@@ -34,7 +37,7 @@
     },
     "extra": {
         "branch-alias": {
-            "dev-master": "1.x-dev"
+            "dev-master": "2.x-dev"
```
Comment (Tobion) [inline composer.json:40]: should be reverted

**Comment by Tobion** [comment]
I think I've changed my mind as the swagger namespace change is quite annoying to support in one version. It's fine to have a different version for swagger-php v3. But since there is no bc break, it can be released as v1.1 and does not require a new major version.

**Comment by TerjeBr** [comment]
Any progress on this? My company are looking to start using OpenAPI v3 now.

**Comment by Tobion** [comment]
@kswzr thanks so much for working on this. I've resolved this in #9.
@TerjeBr I've released https://github.com/Tobion/OpenAPI-Symfony-Routing/releases/tag/v1.2.0 which supports OpenAPI v3.

**Pull request closed without merging**